src/Data/Lens/Index.purs

```diff
@@ -15,6 +15,33 @@ import Data.Set as S
 import Data.StrMap as SM
 import Data.Traversable (traverse)
 
+-- | Use an `Index` optic on types where: 
+-- | 1. The focus element might not be present.
+-- | 2. You either cannot or do not want to add new elements or delete existing ones. 
+-- |
+-- | `Array` is a typical example:
+-- |
+-- | ```purescript 
+-- | preview (ix 1) [0, 1, 2] == Just 1
+-- |
+-- | set (ix 1) 8888 [0, 1, 2] == [0,8888,2]
+-- | ```
+-- |
+-- | Note the use of `preview` rather `view`.
+-- | 
+-- | Another common use is a `Map` that you don't want to either grow or shrink:
+-- |
+-- | ```purescript 
+-- | (set (ix 1) 8888 $ Map.singleton 1 2) == Map.singleton 1 8888
+-- | 
+-- | (set (ix 1) 8888 $ Map.empty) == Map.empty
```

Comment:
Maybe worth using `String` keys here to make the example more obviously distinct from the `Array` one.